apps/search/api/search.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from itertools import chain
+from typing import Optional
+
+from elasticsearch_dsl import Search as DSLSearch
+
+from apps.search.documents.entity import EntityDocument
+from apps.search.documents.image import ImageDocument
+from apps.search.documents.occurrence import OccurrenceDocument
+from apps.search.documents.quote import QuoteDocument
+from apps.search.documents.source import SourceDocument
+
+SEARCHABLE_DOCUMENTS = {
+    OccurrenceDocument.index_name(): OccurrenceDocument,
+    SourceDocument.index_name(): SourceDocument,
+    EntityDocument.index_name(): EntityDocument,
+    QuoteDocument.index_name(): QuoteDocument,
+    ImageDocument.index_name(): ImageDocument,
+}
+
+
+class Search(DSLSearch):
+    results_count: int
+    results_by_id: Optional[dict]
+
+    def to_queryset(self, view):
+        """
+        Resolves elasticsearch results to django models.
+        """
+        s = self
```

Comment:
Would it be OK to just use `self` for readability, or is this necessary to keep original attribute values for `self` or something?

Reply:
We can't just use self because of `s = s.execute()` line, where we need save the response into a variable. I renamed `s` to `response` for better readability (commit pending)